src/badfish/main.py

```diff
@@ -1353,115 +1223,174 @@ async def toggle_boot_device(self, device):
+                if str(_data.get("Inserted")).lower() == "true" and "CD" in str(_data.get("Id")):
+                    inserted = True
             except ValueError:
-                raise BadfishException(
-                    "There was something wrong getting values for VirtualMedia"
-                )
-
+                raise BadfishException("There was something wrong getting values for VirtualMedia")
+        return inserted
+
+    async def mount_virtual_media(self, host, path):
+        vm_config = await self.get_virtual_media_config()
+        _headers = {"Content-Type": "application/json"}
+        if self.vendor == "Supermicro":
+            _payload = {"Host": host, "Path": path, "Username": "", "Password": ""}
+            _uri = "%s%s" % (self.host_uri, vm_config["config"])
+            _response = await self.patch_request(_uri, payload=_payload, headers=_headers)
+
+            _uri = "%s%s/Actions/IsoConfig.Mount" % (self.host_uri, vm_config["config"])
+            _response = await self.post_request(_uri, payload={}, headers=_headers)
+            if _response.status in [200, 202]:
+                self.logger.info("Image mounting operation was successful.")
+            else:
+                raise BadfishException("There was something wrong trying to mount virtual media.")
+        else:
+            vcd = [x for x in vm_config if "CD" in x][0]
+            _uri = "%s%s/Actions/VirtualMedia.InsertMedia" % (self.host_uri, vcd)
+            _payload = {"Image": host + path}
+            _response = await self.post_request(_uri, payload=_payload, headers=_headers)
+            status = _response.status
+            if status == 204:
+                self.logger.info("Image mounting operation was successful.")
+            elif status == 405:
+                self.logger.error("Virtual media mounting is not allowed on this server.")
+                return False
+            elif status == 500:
+                self.logger.error("Couldn't mount virtual media, because there is virtual media mounted already.")
+                return False
+            else:
+                raise BadfishException("There was something wrong trying to mount virtual media.")
         return True
 
     async def unmount_virtual_media(self):
+        vm_config = await self.get_virtual_media_config()
+        _headers = {"Content-Type": "application/json"}
+        if self.vendor == "Supermicro":
+            _uri = "%s%s/Actions/IsoConfig.UnMount" % (
+                self.host_uri,
+                vm_config["config"],
+            )
+            _response = await self.post_request(_uri, payload="{}", headers=_headers)
+            if _response.status in [200, 202]:
+                self.logger.info("Image unmount operation was successful.")
+            else:
+                raise BadfishException("There was something wrong trying to unmount virtual media.")
+            _payload = {"Host": "", "Path": "", "Username": "", "Password": ""}
```

Comment:
is this supposed to be totally empty?

Reply:
Yes, this is supposed to remove any information left from the previously mounted ISO. It doesn't automatically remove the information from the Supermicro's config for virtual media mounting on unmount request.